source/server/hot_restarting_child.cc

```diff
@@ -69,7 +74,11 @@ void HotRestartingChild::sendParentAdminShutdownRequest(time_t& original_start_t
   std::unique_ptr<HotRestartMessage> wrapped_reply = receiveHotRestartMessage(Blocking::Yes);
   RELEASE_ASSERT(replyIsExpectedType(wrapped_reply.get(), HotRestartMessage::Reply::kShutdownAdmin),
                  "Hot restart parent did not respond as expected to ShutdownParentAdmin.");
-  original_start_time = wrapped_reply->reply().shutdown_admin().original_start_time_unix_seconds();
+
+  using u64_secs = std::chrono::duration<uint64_t, std::ratio<1>>;
+  using time_point = std::chrono::time_point<std::chrono::steady_clock, u64_secs>;
+  original_start_time = time_point(
```

Comment:
Consider replacing these 3 lines with:
```
  const uint64_t secs = wrapped_reply->reply().shutdown_admin().original_start_time_unix_seconds();
  original_start_time = MonotonicTime(std::chrono::seconds(secs));
```
but if that doesn't work what you have is ok. I thought the above form might be easier to read but it's up to you.

Reply:
+1, we have MonotonicTime type alias so here we should use it.